pkg/pgmodel/model/custom_types.go

```diff
@@ -0,0 +1,75 @@
+package model
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgtype"
+	"github.com/timescale/promscale/pkg/pgmodel/common/schema"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+var (
+	// Read-only fields after the ingestor inits.
+	labelArrayOID      uint32
+	isLabelArrayOIDSet bool
+)
+
+func labelArrayTranscoder() pgtype.ValueTranscoder { return new(pgtype.Int4Array) }
+
+func RegisterLabelArrayOID(conn pgxconn.PgxConn) error {
+	err := conn.QueryRow(context.Background(), `SELECT '`+schema.Prom+`.label_array'::regtype::oid`).Scan(&labelArrayOID)
+	if err != nil {
+		return fmt.Errorf("registering prom_api.label_array oid: %w", err)
+	}
+	isLabelArrayOIDSet = true
+	return nil
+}
+
+var (
+	// Read-only fields after the ingestor inits.
+	labelValueArrayOID      uint32
+	isLabelValueArrayOIDSet bool
+)
+
+func labelValueArrayTranscoder() pgtype.ValueTranscoder { return new(pgtype.TextArray) }
+
+func RegisterLabelValueArrayOID(conn pgxconn.PgxConn) error {
+	err := conn.QueryRow(context.Background(), `SELECT '`+schema.Prom+`.label_value_array'::regtype::oid`).Scan(&labelValueArrayOID)
+	if err != nil {
+		return fmt.Errorf("registering prom_api.label_value_array oid: %w", err)
+	}
+	isLabelValueArrayOIDSet = true
+	return nil
+}
+
+const (
+	LabelArray = iota
+	LabelValueArray
+)
+
+// GetCustomType returns a custom pgtype.
+func GetCustomType(t uint8) *pgtype.ArrayType {
```

Comment:
TODO: make type more understandable..

Reply:
I think its better to go with the enum approach since it keeps code minimalistic. Imagine we creating more 5 types in future, having functions for each does not seem right. Hence, an enum approach is cleaner.